### ISSUE TITLE
A problem with the new --cors flag

### ISSUE THREAD

**Issue by pavel** [tapio/live-server#113]
@genu posted
> I'm still unable to use the --cors flag it seems. In Chrome requests still get blocked across origins. I have to resort to a chrome extension in order to force cors. I was hoping that --cors would eliminate the need for it, did I misunderstand the purpose of the --cors flag?

You did understand everything right. Seems like you're having a problem. Can you please share a sample of the code you're using (gist or here in a formatted way)? Also please post an output of `npm ls` executed in the project's folder, you're having the trouble with.

**Comment by pavel** [comment]
I've ran some local tests. My results are:
* `http://localhost:8080` -> `http://localhost:8081` :+1:
* `http://localhost:8080` -> `https://localhost:8081` :+1:
* `https://localhost:8080` -> `https://localhost:8081` :+1:
* `https://localhost:8080` -> `http://localhost:8081` :+1:  (an XHR request is blocked by a browser due to [mixed content](https://developer.mozilla.org/en-US/docs/Security/Mixed_content), which is a correct behavior)

**Comment by genu** [comment]
@pavel Hmm, I'm doing ajax calls across origin (i.e. `localhost` to `http://my_server_api`). This may not be an issue, but would the domain difference change the behavior?

In your examples the domain is the same, but different port numbers.

**Comment by pavel** [comment]
@genu Actually, the origin is "host" + "port" so in my examples above AJAX calls are cross-origin.
See https://developer.mozilla.org/en-US/docs/Web/Security/Same-origin_policy, section "Definition of an origin".

So whatever you change, host, port or both, requests become "cross-origin". I still would like to see your example, and `npm ls` output, as requested in my very first comment.

**Comment by genu** [comment]
no, `http://my_server_api` is not a `live-server` instance, and CORS is not enabled on it.

Does CORS need to be enabled on that API server?

**Comment by genu** [comment]
@pavel 
The chrome extension that I use is: https://chrome.google.com/webstore/detail/allow-control-allow-origi/nlfbmbojpeacfghkpbjhddihlkkiljbi

All it seems to do is 
> Adds to response 'Allow-Control-Allow-Origin: *' header

My use case is: I want to run the web application locally with live-server without having to also deploy there server locally. This results in a locally deployed web app interacting with a remotely deployed api server.

**Comment by pavel** [comment]
@genu 
If your API server does not respond with appropriate CORS headers, you can any of:

1) Keep using extension.
2) Use a proxy server that will allow CORS.

But this scenario is out of scope of `--cors` feature.

**Comment by genu** [comment]
ok, I see. 

That's what I wanted to clarify.

Thanks,
We can close this.

**Comment by pavel** [comment]
Please, feel free to close the issue.

**Comment by genu** [comment]
I don't think I'm able to close it since I didn't open it.

**Comment by pavel** [comment]
Agree. Closing.
